services/QuillLMS/engines/evidence/lib/evidence/evidence/check.rb

```diff
@@ -20,44 +22,61 @@ module Check
       optimal: true,
     }
 
-    def self.get_feedback(entry, prompt, previous_feedback)
-      triggered_check = find_triggered_check(entry, prompt, previous_feedback)
+    def self.get_feedback(entry, prompt, previous_feedback, feedback_types=nil)
+      triggered_check = find_triggered_check(entry, prompt, previous_feedback, feedback_types)
 
-      triggered_check&.response || fallback_feedback
+      triggered_check || fallback_feedback
     end
 
     # returns first nonoptimal feedback, and if all are optimal, returns automl feedback
-    def self.find_triggered_check(entry, prompt, previous_feedback)
+    def self.find_triggered_check(entry, prompt, previous_feedback, feedback_types=nil)
       auto_ml_check = nil
-      first_nonoptimal_check = nil
 
-      ALL_CHECKS.each do |check_to_run|
+      checks_to_run(feedback_types).reduce(nil) do |response, check_to_run|
         check = check_to_run.run(entry, prompt, previous_feedback)
 
-        next unless check.success?
+        next response unless check.success?
+
+        break check&.response unless check.optimal?
 
-        auto_ml_check = check if check.auto_ml?
+        auto_ml_check = check&.response if check.auto_ml?
 
-        if !check.optimal?
-          first_nonoptimal_check = check
-          break
-        end
+        auto_ml_check
       end
+    rescue => e
+      Evidence.error_notifier.report(e)
 
-      first_nonoptimal_check || auto_ml_check
+      fallback_feedback(e.message)
     end
 
-    def self.fallback_feedback
+    def self.fallback_feedback(debug=nil)
       @error_rule ||= Rule.find_by(rule_type: Rule::TYPE_ERROR)
-      {
+      feedback = {
         feedback: @error_rule.feedbacks.first&.text || FALLBACK_RESPONSE[:feedback],
         feedback_type: @error_rule.rule_type,
         optimal: @error_rule.optimal,
       }
+
+      return feedback.merge({debug: debug}) if debug
+
+      feedback
     rescue => e
       Evidence.error_notifier.report(e)
 
+      return FALLBACK_RESPONSE.merge({debug: debug}) if debug
+
       FALLBACK_RESPONSE
     end
+
+    def self.checks_to_run(feedback_types)
+      return ALL_CHECKS if feedback_types.nil? || feedback_types.empty?
+
+      qualified_feedback_types = feedback_types.map { |t| "Evidence::Check::#{t}" }
```

Comment:
This will raise an  `uninitialized constant`  when a member of `feedback_types` isn't a member of ALL_CHECKS. Cleaner to compare  `feedback_types`  against `ALL_CHECKS.map(&:to_s)` IMO.